Doctrine/Writer/PaymentMethodWriter.php

```diff
@@ -0,0 +1,162 @@
+    /** @var ModelRepository */
+    private $countryRepository;
+    /** @var DataPersisterInterface */
+    private $dataPersister;
+    /** @var CrudServiceInterface */
+    private $crudService;
+
+    public function __construct(
+        ModelManager $entityManager,
+        PaymentAttributes $paymentAttributes,
+        ModelRepository $paymentRepository,
+        ModelRepository $countryRepository,
+        DataPersisterInterface $dataPersister,
+        CrudServiceInterface $crudService
+    )
+    {
+        $this->entityManager = $entityManager;
+        $this->paymentAttributes = $paymentAttributes;
+        $this->paymentRepository = $paymentRepository;
+        $this->countryRepository = $countryRepository;
+        $this->dataPersister = $dataPersister;
+        $this->crudService = $crudService;
+    }
+
+    public function saveAsShopwarePaymentMethod(
+        PaymentMethod $adyenPaymentMethod,
+        Shop $shop
+    ): ImportResult {
+
+        try {
+            $shops = new ArrayCollection([$shop]);
+            $countries = $this->fetchCountryList();
+
+            $existingPaymentAttribute = $this->paymentAttributes->fetchByAdyenType($adyenPaymentMethod->getType());
+            $existingPaymentMethod = null;
+            if (count($existingPaymentAttribute) !== 0) {
+                $existingPaymentMethod = $this->paymentRepository->findOneBy([
+                    'id' => $existingPaymentAttribute['paymentmeanID']
+                ]);
+            }
+
+            if ($existingPaymentMethod) {
+                $existingPaymentMethod = $existingPaymentMethod->updateFromAdyenPaymentMethod(
+                    $adyenPaymentMethod,
+                    $shops,
+                    $countries
+                );
+                $this->storeAdyenPaymentMethodType(
+                    $existingPaymentMethod->getId(),
+                    $adyenPaymentMethod->getType()
+                );
+
+                return ImportResult::success($shop, $adyenPaymentMethod, true);
+            }
+
+            $shopwarePaymentModel = Payment::createFromAdyenPaymentMethod($adyenPaymentMethod, $shops, $countries);
+
+            $this->entityManager->persist($shopwarePaymentModel);
+            $this->entityManager->flush();
+
+            $this->storeAdyenPaymentMethodType(
+                $shopwarePaymentModel->getId(),
+                $adyenPaymentMethod->getType()
+            );
+
+            return ImportResult::success($shop, $adyenPaymentMethod, false);
+        } catch(\Doctrine\ORM\ORMException $exception) {
+            throw new $exception;
+        } catch(\Exception $exception) {
+            throw new $exception;
+        }
+    }
+
+    private function fetchCountryList(): ArrayCollection
+    {
+        return new ArrayCollection($this->countryRepository->findAll());
+    }
+
+    private function storeAdyenPaymentMethodType(
+        int $paymentMeanId,
+        string $adyenPaymentMethodType
+
+    ) {
```

Comment:
code style clean-up:

```suggestion
    private function storeAdyenPaymentMethodType(
        int $paymentMeanId,
        string $adyenPaymentMethodType
    ) {
```